snapcraft/explanation/parts-lifecycle.md

```diff
@@ -0,0 +1,155 @@
+    * To apply specific permissions to files, you can use the `permissions` keyword.
+    * Some extra checks are also run during this step to ensure that all dependencies are satisfied for a proper run time. If confinement was set to `classic`, then files will be scanned and, if needed, patched to work with this confinement mode.
+1. **Pack**: takes the entire contents of the `prime` directory and packs it into [a snap](/t/the-snap-format/698).
+
+<h3 id='#heading--commands'>Commands<sup><a href=##heading--commands>⚓</a></sup></h3>
+
+Each of these lifecycle steps can be run from the command line, and the command can be part specific or be applied to all parts in a project.
+
+1. `snapcraft pull [<part-name>]`
+1. `snapcraft build [<part-name>]`
+1. `snapcraft stage [<part-name>]`
+1. `snapcraft prime [<part-name>]`
+1. `snapcraft pack` or `snapcraft`
+
+Note that each command also executes the previous lifecycle steps, so `snapcraft` executes all the lifecycle steps chained together.
+
+To access the part environment at any stage, add the `--shell` argument. For example, `snapcraft prime --shell` will run up to the *prime* step and open a shell. See [Iterating over a build](/t/iterating-over-a-build/12143) for more details.
+
+<h3 id='heading--directories'>Directories<sup><a href=#heading--directories>⚓</a></sup></h3>
+
+When running through its lifecycle steps, a part will use different working directories. The directories' names closely follow the steps' names.
+
+| Environment variable | Directory | Purpose |
+|--|--|--|
+| `CRAFT_PART_SRC` | **`parts/<part-name>/src`** | the location of the source after the *pull* step |
+| `CRAFT_PART_BUILD` | **`parts/<part-name>/build`** | the working directory during the *build* step |
+| `CRAFT_PART_INSTALL`| **`parts/<part-name>/install`** | contains the results of the *build* step and the stage packages. It is also the directory where the `organize` event renames the built files |
+| `CRAFT_STAGE` | **`stage`** | shared by all parts, this directory contains the contents of each part's `CRAFT_PART_INSTALL` after the *stage* step. It can contain development libraries, headers, and other components (e.g. pkgconfig files) that need to be accessible from other parts |
+| `CRAFT_PRIME` | **`prime`** | shared by all parts, this directory holds the final components after the *prime* step |
+| `CRAFT_PROJECT_DIR` | path to the current project's subtree in the filesystem. | path to the resulting snap after the *pack* step |
```

Comment:
I'm not sure the "purpose" section is useful information in this context. `$CRAFT_PROJECT_DIR` only exists when snapcraft is running.  But when snapcraft is running, it doesn't matter where the resulting snap will go.

This variable is more used for accessing a resource in your project directory when the lifecycle is running. For example, [`icon: $CRAFT_PROJECT_DIR/resources/icons/128-apps-ghostwriter.png`](https://github.com/KDE/ghostwriter/blob/6cc5988c4bdd713c1126cfa3235427eb6709b660/snapcraft.yaml#L10).

I can't think of a succinct way to write this, maybe something like:
```suggestion
| `CRAFT_PROJECT_DIR` | path to the current project's subtree in the filesystem. | This is used to access resources from the project's subtree, such as an icon or version file.)|
```